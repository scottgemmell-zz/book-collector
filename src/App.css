.App {

}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 10vmin;
  padding: 10px 0;
}

.App-header {
  background-color: #282c34;
  display: flex;
  flex-direction: row;
  font-size: calc(10px + 1vmin);
  color: white;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
