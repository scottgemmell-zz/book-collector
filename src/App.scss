$color-bg: #282c34;

body {
	margin: 0;
	padding: 0;
	font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
	  "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
	  sans-serif;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
}
  
code {
	font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New",
	  monospace;
}
  
.App {
	.App-logo {
		animation: App-logo-spin infinite 20s linear;
		height: 10vmin;
		padding: 10px 0;
	}

	.App-header {
		background-color: $color-bg;
		display: flex;
		flex-direction: row;
		
		h1 {
			color: white;
			font-size: calc(10px + 2vmin);
		}
	}
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
