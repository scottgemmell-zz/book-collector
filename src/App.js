import React, { Component } from "react";
import { Grid } from "react-bootstrap";
import logo from "./logo.svg";
import "./App.scss";
import AppRouter from "./router";

class App extends Component {
	render() {
		return (
			<div className="App">
				<header className="App-header">
					<img src={logo} className="App-logo" alt="logo" />
					<h1>
						Book collector
					</h1>
				</header>
				<main>
					<Grid>
						<AppRouter />
					</Grid>
				</main>
			</div>
		);
	}
}

export default App;
