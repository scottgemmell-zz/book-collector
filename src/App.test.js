import React from "react";
import App from "./App";
import { shallow } from "enzyme";

const defaultProps = () => ({});
const setup = (overrideProps = {}) => {
	const props = Object.assign({}, defaultProps(), overrideProps);
	const wrapper = shallow(<App {...props} />);
	return { wrapper, props };
};

describe("<App />", () => {
	it("renders", () => {
		const { wrapper } = setup();
		expect(wrapper).toMatchSnapshot();
	});
});