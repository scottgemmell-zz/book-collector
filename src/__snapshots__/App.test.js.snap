// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <h1>
            Book collector
          </h1>
        </header>,
        <main>
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <h2>
              Lorem ipsum
            </h2>
            <p>
              dolor set elit nullum vitae.
            </p>
          </Grid>
        </main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <h1>
              Book collector
            </h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Book collector",
            },
            "ref": null,
            "rendered": "Book collector",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <h2>
              Lorem ipsum
            </h2>
            <p>
              dolor set elit nullum vitae.
            </p>
          </Grid>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "container",
            "children": Array [
              <h2>
                Lorem ipsum
              </h2>,
              <p>
                dolor set elit nullum vitae.
              </p>,
            ],
            "componentClass": "div",
            "fluid": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem ipsum",
              },
              "ref": null,
              "rendered": "Lorem ipsum",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "dolor set elit nullum vitae.",
              },
              "ref": null,
              "rendered": "dolor set elit nullum vitae.",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <h1>
              Book collector
            </h1>
          </header>,
          <main>
            <Grid
              bsClass="container"
              componentClass="div"
              fluid={false}
            >
              <h2>
                Lorem ipsum
              </h2>
              <p>
                dolor set elit nullum vitae.
              </p>
            </Grid>
          </main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />,
              <h1>
                Book collector
              </h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Book collector",
              },
              "ref": null,
              "rendered": "Book collector",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Grid
              bsClass="container"
              componentClass="div"
              fluid={false}
            >
              <h2>
                Lorem ipsum
              </h2>
              <p>
                dolor set elit nullum vitae.
              </p>
            </Grid>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "container",
              "children": Array [
                <h2>
                  Lorem ipsum
                </h2>,
                <p>
                  dolor set elit nullum vitae.
                </p>,
              ],
              "componentClass": "div",
              "fluid": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem ipsum",
                },
                "ref": null,
                "rendered": "Lorem ipsum",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "dolor set elit nullum vitae.",
                },
                "ref": null,
                "rendered": "dolor set elit nullum vitae.",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<App /> renders 2 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="App-header"
        >
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <h1>
            Book collector
          </h1>
        </header>,
        <main>
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <h2>
              Lorem ipsum
            </h2>
            <p>
              dolor set elit nullum vitae.
            </p>
          </Grid>
        </main>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <h1>
              Book collector
            </h1>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Book collector",
            },
            "ref": null,
            "rendered": "Book collector",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <h2>
              Lorem ipsum
            </h2>
            <p>
              dolor set elit nullum vitae.
            </p>
          </Grid>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "container",
            "children": Array [
              <h2>
                Lorem ipsum
              </h2>,
              <p>
                dolor set elit nullum vitae.
              </p>,
            ],
            "componentClass": "div",
            "fluid": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Lorem ipsum",
              },
              "ref": null,
              "rendered": "Lorem ipsum",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "dolor set elit nullum vitae.",
              },
              "ref": null,
              "rendered": "dolor set elit nullum vitae.",
              "type": "p",
            },
          ],
          "type": [Function],
        },
        "type": "main",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="App-header"
          >
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <h1>
              Book collector
            </h1>
          </header>,
          <main>
            <Grid
              bsClass="container"
              componentClass="div"
              fluid={false}
            >
              <h2>
                Lorem ipsum
              </h2>
              <p>
                dolor set elit nullum vitae.
              </p>
            </Grid>
          </main>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />,
              <h1>
                Book collector
              </h1>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Book collector",
              },
              "ref": null,
              "rendered": "Book collector",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Grid
              bsClass="container"
              componentClass="div"
              fluid={false}
            >
              <h2>
                Lorem ipsum
              </h2>
              <p>
                dolor set elit nullum vitae.
              </p>
            </Grid>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "container",
              "children": Array [
                <h2>
                  Lorem ipsum
                </h2>,
                <p>
                  dolor set elit nullum vitae.
                </p>,
              ],
              "componentClass": "div",
              "fluid": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Lorem ipsum",
                },
                "ref": null,
                "rendered": "Lorem ipsum",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "dolor set elit nullum vitae.",
                },
                "ref": null,
                "rendered": "dolor set elit nullum vitae.",
                "type": "p",
              },
            ],
            "type": [Function],
          },
          "type": "main",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
